docs/adr/0003-decouple-welcome-email-from-new-user-flow.md

```diff
@@ -0,0 +1,76 @@
+
+## Pros and Cons of the Options
+
+### Option 1: Implement Redis Queue to send welcome email
+
+[Redis Queue](REDIS_QUEUE) would be Relay's first attempt at using Redis worker outside of the Redis session and backend cache with [django-redis](https://pypi.org/project/django-redis/) but can be used to more quickly send the welcome email for new users who join Relay.
+
+- Good, because it reduces the overhead of signal/receiver and delay in responding to Firefox during the `/api/v1/terms-accepted-user` endpoint call.
+- Good, because marketing team has worked on converting their service, Basket, to using this so we could follow their infrstracture and code setup.
+  - Example code snippet using the `@rq_task` decorator [here](https://github.com/mozmeao/basket/blob/main/basket/news/tasks.py) and the command for running the workers [here](https://github.com/mozmeao/basket/blob/main/basket/base/management/commands/rqworker.py).
+- Bad, because [Redis persistence](https://redis.io/docs/management/persistence/) adjacent problems which require infrastructure work.
+
+### Option 2: Utilize Basket API to send welcome email
+
+[Basket API](BASKET), Mozilla's newsletter service for the marketing team, would be a reuse and expansion of Relay's marketing email for users signed up for premium email, phone, and bundle waitlist. Basket handles both transactional and marketing emails. Examples of transactional email is processing primary email address change (transaction) which results in email about the address update. Examples of marketing emails is sending a user about premium subscription being available in their region.
+
+- Good, because Relay has used Basket for waitlists before and this expands on the existing use of Basket.
+- Good, because it is less complex since the problem is handled external to Relay--less code and no new infrastracture needed in Relay to support Option 1.
+- Bad, because Basket is mainly used for marketing purposes and it currently does not have a way to distinguish between transactional and marketing emails.
+- Bad, because it is recommended that the services own their transactional emails.
+
+### Option 3: Implement cron job to send welcome email
+
+Relay already has [many scheduled jobs](https://dashboard.heroku.com/apps/fx-private-relay/scheduler) setup to handle asynchronous work ranging from maintainig old data to renewing monthly phone limits for users.
+
+- Good, because code created to send welcome email can be reused if or when we move to Option 1 using Redis Queue
+- Good, because lowest implementation complexity due to previous use of scheduled jobs
+- Bad, because unlike Redis Queue where more workers can be added to deal with the thundering herd problem, current cron job infrastructure uses set resources.
+
+## Links
+
+[BASKET]: https://basket.readthedocs.io/
+
+- See [Basket PR 1069](https://github.com/mozmeao/basket/pull/1069) for transitioning tasks to use rq
+  [REDIS_QUEUE]: https://python-rq.org/
```

Comment:
I think there is something wrong with this `[REDIS_QUEUE]` bookmark and it isn't parsing correctly:

![Screenshot 2023-09-06 at 1 55 27 PM](https://github.com/mozilla/fx-private-relay/assets/557895/0786bf05-ac9e-4c28-bdf8-407d328987c9)

Reply:
Resolved in c42750f34a51ce9c5e51a7a5754dbdbaae1de9d2